atomicwrites/__init__.py

```diff
@@ -19,12 +19,27 @@ def _path_to_unicode(x):
 
 
 if sys.platform != 'win32':
+    def _sync_directory(directory):
+        # Ensure that filenames are written to disk
+        fd = os.open(directory, 0)
+        try:
+            os.fsync(fd)
+        finally:
+            os.close(fd)
+
     def _replace_atomic(src, dst):
         os.rename(src, dst)
+        _sync_directory(os.path.dirname(dst))
 
     def _move_atomic(src, dst):
         os.link(src, dst)
         os.unlink(src)
```

Comment:
> ``_move_atomic`` errors when ``dst`` exists, ``_replace_atomic`` doesn't.

It doesn't care if ``dst`` exists or not. (https://docs.python.org/3.5/library/os.html#os.rename) There are the ``RENAME_EXCHANGE`` and ``RENAME_NOREPLACE`` flags to control the behaviour, but the python function doesn't expose that.